### ISSUE TITLE
support node v18+

### ISSUE THREAD

**Issue by iambumblehead** [rethinkdb/rethinkdb-ts#101]
when using node v18.1.0, which isn't supported by rethinkdb-ts, this error occurs during connection,
```javascript
/app/node_modules/rethinkdb-ts/lib/response/cursor.js:20
        this.closed = false;
                    ^

TypeError: Cannot set property closed of #<Readable> which has only a getter
    at new Cursor (/app/node_modules/rethinkdb-ts/lib/response/cursor.js:20:21)
    at RethinkDBConnection.query (/app/node_modules/rethinkdb-ts/lib/connection/connection.js:167:16)
    at ServerConnectionPool.queue (/app/node_modules/rethinkdb-ts/lib/connection/server-pool.js:123:44)
    at MasterConnectionPool.queue (/app/node_modules/rethinkdb-ts/lib/connection/master-pool.js:178:21)
    at Function.run (/app/node_modules/rethinkdb-ts/lib/query-builder/query-runner.js:26:67)
```

**Comment by iambumblehead** [comment]
I tried replacing cursor.js' `this.closed = false` with `this.close();` to see what would happen,
```javascript
RangeError: Maximum call stack size exceeded
```

**Comment by atassis** [comment]
That wont help, `close` is our method, there is a `destroy` method of `Readable` which should be called. Will test a fix soon, though dont consider v18 branch reliable until it goes into LTS state
Still good to know about issues, just my personal consideration about production apps=)

**Comment by atassis** [comment]
Just making a note that I do remember about this issue. Thinking about making any kind of a changelog and release 3.0 version of redb-ts, and after that fix current issue

**Comment by iambumblehead** [comment]
maybe remove the `[david | no longer available]` badge from the README

**Comment by atassis** [comment]
It is better to fix it, but also a point

**Comment by atassis** [comment]
Writing here just to inform that I work on this issue now. Smth changed in the way Readable is working, maybe they fixed a bug, which breaks the code now.

**Comment by atassis** [comment]
published `2.5.0`, it passed tests, but I'ld test it still. If smth fails- better to stay at previous point

**Comment by atassis** [comment]
I am closing an issue, if you find something- you can create a new one
